core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -395,24 +399,69 @@ public Catalog createCatalog(
     }
   }
 
-  private Pair<Map<String, String>, Map<String, String>> getCatalogAlterProperty(
-      CatalogChange... catalogChanges) {
-    Map<String, String> upserts = Maps.newHashMap();
-    Map<String, String> deletes = Maps.newHashMap();
+  /**
+   * Test whether a catalog can be created with the specified parameters, without actually creating
+   * it.
+   *
+   * @param ident The identifier of the catalog to be tested.
+   * @param type the type of the catalog.
+   * @param provider the provider of the catalog.
+   * @param comment the comment of the catalog.
+   * @param properties the properties of the catalog.
+   */
+  @Override
+  public void testConnection(
+      NameIdentifier ident,
+      Catalog.Type type,
+      String provider,
+      String comment,
+      Map<String, String> properties) {
+    NameIdentifier metalakeIdent = NameIdentifier.of(ident.namespace().levels());
+    try {
+      if (!store.exists(metalakeIdent, EntityType.METALAKE)) {
+        throw new NoSuchMetalakeException(METALAKE_DOES_NOT_EXIST_MSG, metalakeIdent);
+      }
 
-    Arrays.stream(catalogChanges)
-        .forEach(
-            catalogChange -> {
-              if (catalogChange instanceof SetProperty) {
-                SetProperty setProperty = (SetProperty) catalogChange;
-                upserts.put(setProperty.getProperty(), setProperty.getValue());
-              } else if (catalogChange instanceof RemoveProperty) {
-                RemoveProperty removeProperty = (RemoveProperty) catalogChange;
-                deletes.put(removeProperty.getProperty(), removeProperty.getProperty());
-              }
-            });
+      if (store.exists(ident, EntityType.CATALOG)) {
+        throw new CatalogAlreadyExistsException("Catalog %s already exists", ident);
```

Comment:
Will it lead to test error?

Reply:
yes, whether the catalog name already exists is also a validation item.

Reply:
More precisely, this will cause the `testConnection` method to throw a `CatalogAlreadyExistsException` exception.